edx_proctoring/backends/software_secure.py

```diff
@@ -147,186 +151,31 @@ def on_review_callback(self, payload):
-        review = ProctoredExamSoftwareSecureReview.get_review_by_attempt_code(attempt_code)
-
-        if review:
-            if not constants.ALLOW_REVIEW_UPDATES:
-                err_msg = (
-                    'We already have a review submitted from SoftwareSecure regarding '
-                    'attempt_code {attempt_code}. We do not allow for updates!'.format(
-                        attempt_code=attempt_code
-                    )
-                )
-                raise ProctoredExamReviewAlreadyExists(err_msg)
-
-            # we allow updates
-            warn_msg = (
-                'We already have a review submitted from SoftwareSecure regarding '
-                'attempt_code {attempt_code}. We have been configured to allow for '
-                'updates and will continue...'.format(
-                    attempt_code=attempt_code
-                )
-            )
-            log.warn(warn_msg)
-        else:
-            # this is first time we've received this attempt_code, so
-            # make a new record in the review table
-            review = ProctoredExamSoftwareSecureReview()
-
-        review.attempt_code = attempt_code
-        review.raw_data = json.dumps(payload)
-        review.review_status = review_status
-        review.student = attempt_obj.user
-        review.exam = attempt_obj.proctored_exam
-        # set reviewed_by to None because it was reviewed by our 3rd party
-        # service provider, not a user in our database
-        review.reviewed_by = None
-
-        review.save()
-
-        # go through and populate all of the specific comments
-        for comment in payload.get('webCamComments', []):
-            self._save_review_comment(review, comment)
-
-        for comment in payload.get('desktopComments', []):
-            self._save_review_comment(review, comment)
-
-        # we could have gotten a review for an archived attempt
-        # this should *not* cause an update in our credit
-        # eligibility table
-        if not is_archived_attempt:
-            # update our attempt status, note we have to import api.py here because
-            # api.py imports software_secure.py, so we'll get an import circular reference
-
-            allow_rejects = not constants.REQUIRE_FAILURE_SECOND_REVIEWS
-
-            self.on_review_saved(review, allow_rejects=allow_rejects)
-
-        # emit an event for 'review_received'
-        data = {
-            'review_attempt_code': review.attempt_code,
-            'review_status': review.review_status,
+        SoftwareSecureReviewStatus.validate(payload['reviewStatus'])
+        review_status = SoftwareSecureReviewStatus.to_standard_status.get(payload['reviewStatus'], None)
+
+        comments = []
+        converted = {
+            'status': review_status,
+            'comments': comments,
+            'payload': payload
         }
+        for comment in payload.get('webCamComments', []) + payload.get('desktopComments', []):
```

Comment:
I see that this will work fine, but it will be clearer if you build `comments` fully before using it in `converted`.